include/seqan3/core/simd/view_to_simd.hpp

```diff
@@ -0,0 +1,719 @@
+ */
+template <std::ranges::View urng_t, Simd simd_t>
+class view_to_simd<urng_t, simd_t>::iterator_type
+{
+public:
+    /*!\name Associated types
+     * \{
+     */
+    using reference = chunk_type &; //!< The reference type.
+    using value_type = chunk_type; //!< The value type.
+    using pointer = chunk_type *; //!< The pointer type.
+    using difference_type = ptrdiff_t; //!< The difference type.
+    using iterator_category = std::input_iterator_tag; //!< The iterator category.
+    using iterator_concept = iterator_category; //!< The iterator concept.
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr iterator_type()                                  = default; //!< Defaulted.
+    constexpr iterator_type(iterator_type const &)             = default; //!< Defaulted.
+    constexpr iterator_type(iterator_type &&)                  = default; //!< Defaulted.
+    constexpr iterator_type & operator=(iterator_type const &) = default; //!< Defaulted.
+    constexpr iterator_type & operator=(iterator_type &&)      = default; //!< Defaulted.
+    ~iterator_type()                                           = default; //!< Defaulted.
+
+    /*!\brief Construction from the associated range.
+     * \param this_view A reference to the associated view.
+     *
+     * \details
+     *
+     * Initialises the iterators and sentinels of the underlying sequences to be transformed and calls
+     * underflow to fetch the first chunk.
+     */
+    constexpr iterator_type(view_to_simd & this_view) : this_view{&this_view}, cached_chunk_pos{0}
+    {
+        // Initialise the iterator of the sub ranges.
+        size_t seq_id = 0;
+        for (auto it = std::ranges::begin(this_view.urng); it != std::ranges::end(this_view.urng); ++it, ++seq_id)
+        {
+            cached_iter[seq_id] = std::ranges::begin(*it);
+            cached_sentinel[seq_id] = std::ranges::end(*it);
+        }
+
+        // If the batch is not full, i.e. less than chunk_size many sequences, then fill them up with dummy sequences.
+        for (; seq_id < chunk_size; ++seq_id)
+        {
+            cached_iter[seq_id] = std::ranges::begin(this_view.empty_inner_range);
+            cached_sentinel[seq_id] = std::ranges::end(this_view.empty_inner_range);
+        }
+
+        // Check if this is the final chunk already.
+        final_chunk = all_iterators_reached_sentinel(std::make_index_sequence<chunk_size>{});
+
+        // Fetch the next available input characters from the sequences and transform them into simd vectors.
+        underflow();
+    }
+    //!\}
+
+    /*!\name Element access
+     * \{
+     */
+    //!\brief Returns a reference to the current chunk of simd vectors.
+    constexpr reference operator*() const noexcept
+    {
+        assert(this_view != nullptr);
+        return this_view->cached_simd_chunks[cached_chunk_pos];
```

Comment:
why not 
```suggestion
        return this_view->cached_simd_chunks[current_chunk_pos];
```

instead of cached?